ElectronicsCrashCourse.md

```diff
@@ -0,0 +1,241 @@
+In practice, we can draw more for a short period of time.
+If you are interested in more details about how and why that works, look at the electrical coursework for the team.
+
+The PDH is also usually at one end of our CAN network. What's CAN? Glad you asked . . .
+
+### The CAN Network
+
+![CAN Wire](Assets/CANWire.png)
+
+CAN is a type of communication along wires that allow our sensors and motors to communicate along our robot.
+In FRC, CAN is transmitted over yellow and green pairs of cables.
+Each device on the CAN network has a unique ID that it uses to talk to other devices on the network.
+All of our motors and many of our sensors will communicate over CAN.
+CAN is also one of the easiest places for electrical problems to become apparent in software because of the thin wires and large reach of the system.
+When you start up the robot you may see red error messages in the drive station console reporting CAN errors.
+If you see that, something might be broken on the CAN network.
+
+We have to set the IDs of the devices on the CAN network to make sure they are all uniquely and correctly identified.
+It would be very awkward to try to drive the robot and accidentally run the elevator!
+To set the IDs of devices on the CAN network we use the [Tuner X](https://pro.docs.ctr-electronics.com/en/stable/docs/tuner/index.html) app built by CTRE.
+This app also has features to test and configure devices on the CAN network.
+
+### RoboRIO 2
+
+![RoboRIO](Assets/RoboRIO.jpg)
+
+The RoboRIO (rio) is the brain of the robot.
+It is built around a computer running Linux with a large number of Input/Output (IO) ports.
+These ports include:
+
+- Digital IO (DIO) can accept inputs and outputs that are either a 1 or 0, on or off.
+ When used as an input the signal can rapidly be turned on and off to send numerical data.
+ This is done using Pulse Width Modulation (PWM), essentially measuring how long a signal is on compared to how long it is off in a given time window to get a numerical value.
+ For more information on PWM read [this article](https://learn.sparkfun.com/tutorials/pulse-width-modulation/all)
+- PWM pins provide additional pins to *output* PWM signals.
+  Unlike the DIO ports the PWM ports cannot take inputs.
+  Many motor controllers support using a PWM signal for control instead of CAN, although it has significantly limited features.
+- Analog inputs provide adititional sensor ports, although most sensors use DIO or CAN.
+- The large set of pins in the middle of the Rio is the MXP port.
+  MXP (and the SPI port in the top-right corner) is used for communication over serial interfaces such as I²C protocal.
+  Unfortunately, there is an issue with I²C that can cause the code to lock up when it is used, so we avoid using the serial ports.
+- The CAN network originates at the RIO.
+- Several USB ports are available on the Rio.
+  Common uses include external USB sticks to store logs or the CANivore (see later in this page)
```

Comment:
It's also an option for networking. Probably worth linking to the CANivore section. [Github flavored markdown example of that here.](https://gist.github.com/asabaylus/3071099?permalink_comment_id=3565887#gistcomment-3565887)